src/gluonts/dataset/split/splitter.py

```diff
@@ -250,13 +249,24 @@ def rolling_split(
                 )
                 prediction_length = getattr(self, "prediction_length")
 
-                assert (
-                    train.end + train.end.freq * prediction_length <= test.end
+                self._check_split_length(
+                    train.end, test.end, prediction_length
                 )
                 split._add_test_slice(test)
 
         return split
 
+    @staticmethod
```

Comment:
Not sure I like being a `staticmethod`. I prefer having a standalone function instead.